Code/Framework/AzCore/CMakeLists.txt

```diff
@@ -6,11 +6,15 @@
 #
 #
 
+list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/3rdParty)
+set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH})
+
```

Comment:
Removing this should cause it to not be able to find the FindMimalloc.cmake file?   Note that it only re-finds it if the cache is clean.  So it could be "working for you" with a dirty cache.

When o3de's code sees a BUILD_DEPENDENCY on `3rdParty::xxxxx` and that target does not already exist, it automatically calls `find_package(xxxxx)`.  CMake responds builtin function by searching its `CMAKE_MODULE_PATH` list directories for any file called `Findxxxxx.cmake` and executing it.  CMAKE_MODULE_PATH does not contain the current directory.  It would imply that removing these two lines would cause trouble here.  Is it still working anyway?  How?